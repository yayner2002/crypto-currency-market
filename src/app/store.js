import { configureStore } from "@reduxjs/toolkit";
import coinsReducer from "../redux/crypto/Cryptos";

export const store = configureStore({
  reducer: {
    coins: coinsReducer,
  },
});
