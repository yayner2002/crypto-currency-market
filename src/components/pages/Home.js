import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Container, Row, Col } from 'reactstrap';
import { Link } from 'react-router-dom';
import { fetchCoinInfo, selectCoins } from '../../redux/crypto/Cryptos';
import Helmet from '../Helmet/Helmet';

let init = true;
const Home = () => {
  const coins = useSelector(selectCoins);
  // const {
  //   name, rank, symbol, priceUsd, marketCapUsd,
  // } = coins[0];

  const dispatch = useDispatch();

  useEffect(() => {
    if (init) {
      dispatch(fetchCoinInfo());
      init = false;
    }
  }, [dispatch]);
  return (
    <Helmet title="Home">
      <h1>
        <ul>
          {coins.map((coins) => (
            <li key={coins.id}>{coins.name}</li>
          ))}
        </ul>
      </h1>
    </Helmet>
  );
};

export default Home;
