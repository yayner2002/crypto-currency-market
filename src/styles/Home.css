.coinContainer {
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 0.08rem;
  box-sizing: border-box;
  overflow: hidden;
  background: #6a88be;
  width: 375px;
  margin: auto;
}

a {
  text-decoration: none;
  color: #fff;
}

a:hover {
  color: #fff;
}

.singleCoin {
  background: #305aa8;
  color: #fff;
  display: flex;
  flex-direction: column;
  padding: 0% 0.5rem;
}

.singleCoin span a {
  font-size: 18px;
}

.singleCoin:hover {
  transform: scale(0.9);
  transition: 0.5s;
}

.coins-title {
  color: #fff;
  font-size: 18px;
}

.coin-title-container {
  width: 375px;
  margin: auto;
  text-align: center;
  padding: 0.3rem 0;
  background: #4369b2;
  margin-bottom: 0%;
  border-bottom: 1px solid #325ba7;
  border-top: 1px solid #305aa8;
  display: flex;
  justify-content: space-between;
  flex: 1;
}

.form {
  border: 1px solid #fff;
  border-radius: 1rem;
}

.form input {
  background: transparent;
  border: none;
  outline: none;
  width: 155px;
  color: #fff;
  padding-left: 0.5rem;
}

.form button {
  outline: none;
  border: none;
  background: transparent;
}

.form button i {
  color: #fff;
}

.form input::placeholder {
  color: #fff;
  padding-left: 0.5rem;
}

.singleCoin p {
  padding: 0%;
  margin: 0%;
  font-size: 18px;
  font-weight: 600;
}

.singleCoin span {
  align-self: flex-end;
}
