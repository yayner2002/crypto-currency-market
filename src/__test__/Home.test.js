/* eslint-disable no-unused-vars */
import { toBeInTheDocument } from '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../app/store';

import Home from '../components/pages/Home';

const MockHome = () => (
  <Provider store={store}>
    <Router>
      <Home />
    </Router>
  </Provider>
);

describe('home', () => {
  test('should show texts', () => {
    render(<MockHome />);
    const tree = renderer.create(<MockHome />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(screen.getByTestId('heading')).toBeInTheDocument();
  });
});
