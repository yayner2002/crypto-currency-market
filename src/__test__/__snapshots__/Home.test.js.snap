// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home should show texts 1`] = `
Array [
  <div
    className="show-case-section"
  >
    <div
      className="show-case-image"
    >
      <img
        alt="showcase"
        className="showcaseimg"
        src="showcase1.png"
      />
    </div>
    <div
      className="show-case-title"
    >
      <span>
        <h6>
          Digital Currency
        </h6>
      </span>
      <span>
        <h6>
          0
           
          coins
        </h6>
      </span>
    </div>
  </div>,
  <div
    className="coin-title-container"
  >
    <h5
      className="coins-title"
      data-testid="heading"
    >
      Coins By Name
    </h5>
  </div>,
  <div
    className="coinContainer"
    data-testid="coins-page"
  />,
]
`;
